### PR TITLE
Distributed toilet

### PR DESCRIPTION
**Pull request by varsill**
Description of the problem: https://membraneframework.atlassian.net/browse/MC-57
This PR adds Membrane.Core.Element.Toilet.DistributionCounter - an abstraction put above the implementation of atomic counter.

### CODE REVIEW

File: lib/membrane/core/element/toilet.ex [hunk 1]
```diff
@@ -7,35 +7,125 @@ defmodule Membrane.Core.Element.Toilet do
...
+    @spec new(:same_node | :different_nodes) :: t
+    def new(type) do
+      atomic_ref = :atomics.new(1, [])
+
+      case type do
+        :same_node ->
+          {:same_node, atomic_ref}
+
+        :different_nodes ->
+          {:ok, pid} = GenServer.start(Membrane.Core.Element.Toilet.DistributedCounter.Worker, [])
```
Comment (mat-hek) [inline lib/membrane/core/element/toilet.ex:57]:
```suggestion
          {:ok, pid} = GenServer.start(Worker, [])
```

File: test/membrane/core/toilet_test.exs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+defmodule Membrane.Core.ToiletTest do
```
Comment (mat-hek) [inline test/membrane/core/toilet_test.exs:1]:
```suggestion
defmodule Membrane.Core.Element.ToiletTest do
```

File: lib/membrane/core/element/toilet.ex (hunk 1 above)
Comment (varsill) [reply lib/membrane/core/element/toilet.ex:57]: done

File: test/membrane/core/toilet_test.exs (hunk 2 above)
Comment (varsill) [reply test/membrane/core/toilet_test.exs:1]: done and the file has been moved to the appropriate directory

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane/core/element/demand_handler.ex [hunk 3]
```diff
@@ -123,23 +122,22 @@ defmodule Membrane.Core.Element.DemandHandler do
...
-      :delay ->
+      {:delay, toilet} ->
         PadModel.set_data!(
           state,
           pad_ref,
-          :unrinsed_buffers_size,
-          unrinsed_buffers_size + buf_size
+          :toilet,
+          toilet
         )
```
[the remark this answers is not in the document]
Comment (varsill) [reply lib/membrane/core/element/demand_handler.ex]: Sure, it's just a leftover of the previous implementation.
done

**Review by mat-hek — APPROVED** [review]

File: lib/membrane/core/playback_handler.ex [hunk 4]
```diff
@@ -122,8 +122,11 @@ defmodule Membrane.Core.PlaybackHandler do
         continue_playback_change(handler, state)
       end
     else
-      {:error, reason} -> {{:error, reason}, state}
-      other_error -> other_error
+      {:error, reason} ->
+        {{:error, reason}, state}
+
+      other_error ->
+        other_error
```
[the remark this answers is not in the document]
Comment (varsill) [reply lib/membrane/core/playback_handler.ex]: i have reverted them

**Pull request merged**